util/quarantine/quarantine_darwin.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package quarantine
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"golang.org/x/sys/unix"
+)
+
+func setQuarantineAttr(f *os.File) error {
+	sc, err := f.SyscallConn()
+	if err != nil {
+		return err
+	}
+
+	now := time.Now()
+
+	// We uppercase the UUID to match what other applications on macOS do
+	id := strings.ToUpper(uuid.New().String())
+
+	attrData := fmt.Sprintf("%s;%x;%s;%s",
```

Comment:
Pointer to documentation of this format? https://eclecticlight.co/2020/10/29/quarantine-and-the-quarantine-flag/ is the best that I've found, but there may be others.

Reply:
Good call; added